packages/backend/src/open_payments/receiver/service.ts

```diff
@@ -0,0 +1,212 @@
+import {
+  OpenPaymentsClient,
+  IncomingPayment as OpenPaymentsIncomingPayment,
+  ILPStreamConnection as OpenPaymentsConnection
+} from 'open-payments'
+
+import { parseAmount } from '../amount'
+import { ConnectionService } from '../connection/service'
+import { PaymentPointerService } from '../payment_pointer/service'
+import { BaseService } from '../../shared/baseService'
+import { IncomingPaymentService } from '../payment/incoming/service'
+import { PaymentPointer } from '../payment_pointer/model'
+import { Receiver } from './model'
+
+// A receiver is resolved from an incoming payment or a connection
+export interface ReceiverService {
+  get(url: string): Promise<Receiver | undefined>
+}
+
+interface ServiceDependencies extends BaseService {
+  accessToken: string
+  connectionService: ConnectionService
+  incomingPaymentService: IncomingPaymentService
+  openPaymentsUrl: string
+  paymentPointerService: PaymentPointerService
+  openPaymentsClient: OpenPaymentsClient
+}
+
+const CONNECTION_URL_REGEX = /\/connections\/(.){36}$/
+const INCOMING_PAYMENT_URL_REGEX =
+  /(?<paymentPointerUrl>^(.)+)\/incoming-payments\/(?<id>(.){36}$)/
+
+export async function createReceiverService(
+  deps_: ServiceDependencies
+): Promise<ReceiverService> {
+  const log = deps_.logger.child({
+    service: 'ReceiverService'
+  })
+  const deps: ServiceDependencies = {
+    ...deps_,
+    logger: log
+  }
+
+  return {
+    get: (url) => getReceiver(deps, url)
+  }
+}
+
+async function getReceiver(
+  deps: ServiceDependencies,
+  url: string
+): Promise<Receiver | undefined> {
+  if (url.match(CONNECTION_URL_REGEX)) {
+    const connection = await getConnection(deps, url)
+    if (connection) {
+      return Receiver.fromConnection(connection)
+    }
+  } else {
+    const incomingPayment = await getIncomingPayment(deps, url)
+    if (incomingPayment) {
+      return Receiver.fromIncomingPayment(incomingPayment)
+    }
+  }
+}
+
+async function getLocalConnection(
+  deps: ServiceDependencies,
+  url: string
+): Promise<OpenPaymentsConnection | undefined> {
+  const incomingPayment = await deps.incomingPaymentService.getByConnection(
+    url.slice(-36)
+  )
+
+  if (!incomingPayment) {
+    return
+  }
+
+  const connection = deps.connectionService.get(incomingPayment)
+
+  if (!connection) {
+    return
+  }
+
+  return connection.toOpenPaymentsType()
+}
+
+async function getConnection(
+  deps: ServiceDependencies,
+  url: string
+): Promise<OpenPaymentsConnection | undefined> {
+  try {
+    if (url.startsWith(`${deps.openPaymentsUrl}/connections/`)) {
+      return await getLocalConnection(deps, url)
+    }
+
+    return await deps.openPaymentsClient.ilpStreamConnection.get({
+      url,
+      accessToken: deps.accessToken
+    })
+  } catch (_) {
+    return undefined
+  }
+}
+
+function parseIncomingPaymentUrl(
```

Comment:
What do you think about this living in `open-payments` as something like `client.incomingPayment.parseUrl`?

Reply:
I think I could be swayed to put it in the client later on, especially that it's the one that knows about routes... for now this helper is ok here IMO